post.js

```diff
@@ -0,0 +1,163 @@
+async function loadPost(url) {
+  try {
+    const response = await fetch(url);
+    if (!response.ok) {
+      throw new Error(`HTTP error! Status: ${response.status}`);
+    }
+    return await response.text();
+  } catch (error) {
+    console.error("Error fetching post:", error);
+  }
+}
+
+async function parsePostHtml(htmlContent) {
+  const parser = new DOMParser();
+  const doc = parser.parseFromString(htmlContent, "text/html");
+  const alternateLink = doc.querySelector('link[rel="alternate"]');
+  return alternateLink ? alternateLink.href : null;
+}
+
+async function fetchJsonLd(jsonLdUrl) {
+  try {
+    const response = await fetch(jsonLdUrl);
+    if (!response.ok) {
+      throw new Error(`HTTP error! Status: ${response.status}`);
+    }
+    return await response.json();
+  } catch (error) {
+    console.error("Error fetching JSON-LD:", error);
+  }
+}
+
+function renderPost(jsonLdData) {
```

Comment:
We should make this a method in the post component. Since IDs are unique to a document we should either use the shadow DOM, class names, or better yet construct the DOM tree with `document.createElement` or better yet with template elements: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/template